python/cuxfilter/charts/datashader/custom_extensions/holoviews_datashader.py

```diff
@@ -0,0 +1,577 @@
+    pixel_shade_type = param.String("linear")
+    box_stream = param.ClassSelector(
+        class_=hv.streams.SelectionXY, default=hv.streams.SelectionXY()
+    )
+    lasso_stream = param.ClassSelector(
+        class_=hv.streams.Lasso, default=hv.streams.Lasso()
+    )
+    reset_stream = param.ClassSelector(
+        class_=hv.streams.PlotReset,
+        default=hv.streams.PlotReset(resetting=False),
+    )
+    spread_threshold = param.Number(
+        0, doc="threshold parameter passed to dynspread function"
+    )
+    tile_provider = param.String(None)
+
+    @property
+    def vdims(self):
+        if self.aggregate_col is None:
+            return [self.y]
+        return [self.y, self.aggregate_col]
+
+    def __init__(self, **params):
+        """
+        initialize InteractiveDatashader object, and set a listener on
+        self.data
+        """
+        super(InteractiveDatashader, self).__init__(**params)
+        self.tiles = (
+            tile_sources[self.tile_provider]()
+            if (self.tile_provider is not None)
+            else self.tile_provider
+        )
+
+    def add_box_select_callback(self, callback_fn):
+        self.box_stream = hv.streams.SelectionXY(subscribers=[callback_fn])
+
+    def add_lasso_select_callback(self, callback_fn):
+        self.lasso_stream = hv.streams.Lasso(subscribers=[callback_fn])
+
+    def reset_all_selections(self):
+        self.lasso_stream.reset()
+        self.box_stream.reset()
+
+    def add_reset_event(self, callback_fn):
+        self.reset_stream = hv.streams.PlotReset(subscribers=[callback_fn])
+
+
+class InteractiveDatashaderPoints(InteractiveDatashader):
+    aggregate_col = param.String(allow_None=True)
+    aggregate_fn = param.String("count")
+    legend = param.Boolean(True, doc="whether to display legends or not")
+    legend_position = param.String("right", doc="position of legend")
+    cmap = param.Dict(default={"cmap": CUXF_DEFAULT_COLOR_PALETTE})
+    tools = param.List(
+        default=["pan", "reset", "lasso_select", "wheel_zoom"],
+        doc="interactive tools to add to the chart",
+    )
+    color_palette = param.List()
+    point_shape = param.ObjectSelector(
+        default="circle",
+        objects=[
+            "circle",
+            "square",
+            "rect_vertical",
+            "rect_horizontal",
+            "cross",
+        ],
+    )
+    max_px = param.Integer(10)
+    clims = param.Tuple(default=(None, None))
+
+    def __init__(self, **params):
+        super(InteractiveDatashaderPoints, self).__init__(**params)
+        self._compute_datashader_assets()
+
+    def _compute_clims(self):
+        if not isinstance(
+            self.source_df[self.aggregate_col].dtype,
+            cudf.core.dtypes.CategoricalDtype,
+        ):
+            self.clims = (
+                self.source_df[
+                    self.aggregate_col
+                    if self.aggregate_col is not None
+                    else self.y
```

Comment:
Unless  `self.aggregate_col` can otherwise be falsey, then perhaps
```suggestion
                    self.aggregate_col or self.y
```
here and below? Could also factor out this column name into a variable above so that it is not computed identically in two places.

Reply:
fixed, actually it was taken care of in the [initialization for core_non_aggregate class](https://github.com/AjayThorve/cuxfilter/blob/fix/remove-custom-extensions/python/cuxfilter/charts/core/non_aggregate/core_scatter.py#L87), so we could directly use self.aggregate_col here